subsys/bluetooth/mesh/cfg_cli.c

```diff
@@ -2064,3 +2067,68 @@ void bt_mesh_cfg_cli_timeout_set(int32_t timeout)
 {
 	msg_timeout = timeout;
 }
+
+int bt_mesh_comp_p0_get(struct bt_mesh_comp_p0 *page,
+			struct net_buf_simple *buf)
+{
+	if (buf->len < 10) {
+		return -EINVAL;
+	}
+
+	page->cid = net_buf_simple_pull_le16(buf);
+	page->pid = net_buf_simple_pull_le16(buf);
+	page->vid = net_buf_simple_pull_le16(buf);
+	page->crpl = net_buf_simple_pull_le16(buf);
+	page->feat = net_buf_simple_pull_le16(buf);
+	page->_buf = buf;
+
+	return 0;
+}
+
+struct bt_mesh_comp_p0_elem *bt_mesh_comp_p0_elem_pull(const struct bt_mesh_comp_p0 *page,
+						       struct bt_mesh_comp_p0_elem *elem)
+{
+	size_t modlist_size;
+
+	if (page->_buf->len < 4) {
+		return NULL;
+	}
+
+	elem->loc = net_buf_simple_pull_le16(page->_buf);
+	elem->nsig = net_buf_simple_pull_u8(page->_buf);
+	elem->nvnd = net_buf_simple_pull_u8(page->_buf);
+
+	modlist_size = elem->nsig * 2 + elem->nvnd * 4;
+
+	if (page->_buf->len < modlist_size) {
+		return NULL;
+	}
+
+	elem->_buf = net_buf_simple_pull_mem(page->_buf, modlist_size);
+
+	return elem;
+}
+
+uint16_t bt_mesh_comp_p0_elem_mod(struct bt_mesh_comp_p0_elem *elem, int idx)
+{
+	if (idx >= elem->nsig) {
+		return 0xffff;
+	}
```

Comment:
Don't we want to take into use `CHECKIF()` from `include/sys/check.h`? I've seen that used in many Bluetooth audio related patches. It seems the general trend is to move toward that for all input parameter checking.

Reply:
I wasn't aware of this. Makes sense, I'll change it.